### PR TITLE
Feat/basic auth extension

### PR DESCRIPTION
**Pull request by ChristophMarotzke**
Basic Authentication extensions for CAP (custom Authentication endpoint(s))

### CODE REVIEW

**Comment by ChristophMarotzke** [comment]
@Siolto We have created Pull request against "main" Is this correct? Kindly request for review.

**Comment by Siolto** [comment]
> @Siolto We have created Pull request against "main" Is this correct? Kindly request for review.
yep this is correct :)

File: src/types/wdi5.types.ts [hunk 1]
```diff
@@ -86,6 +86,7 @@ export type BTPAuthenticator = {
 
 export type BasicAuthAuthenticator = {
     provider: "BasicAuth"
+    basicAuthUrl?: Array<string>
```
Comment (Siolto) [inline src/types/wdi5.types.ts:89]: @vobu @monavari-lebrecht: basicAuthUrl is an Array. Should we rename it to `basicAuthUrls`? I am not sure. Sound a little bit awkward 😄

Comment (vobu) [reply src/types/wdi5.types.ts:89]: yep, do it!

File: src/lib/authentication/BasicAuthenticator.ts [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
-    constructor(browserInstanceName) {
+    private options
+    private baseUrl
+
+    constructor(options: BasicAuthenticatorType, browserInstanceName, baseUrl: string) {
         super(browserInstanceName)
+        this.options = options
+        this.baseUrl = baseUrl
     }
     async login() {
```
Comment (mpah) [inline src/lib/authentication/BasicAuthenticator.ts:13]: To not copy paste the code & regex we could do something like.

```
 const basicAuthUrls = this.sOptions.basicAuthUrls || [this.baseUrl]
        for (const url of basicAuthUrls ) {
             //rest of the code or better separate private method loginBasicUrl(url: string) doing the work
        }
```

**Review by mpah — CHANGES_REQUESTED** [review]

**Pull request merged**